ext/intl/dateformat/dateformat_format_object.cpp

```diff
@@ -158,28 +158,28 @@ U_CFUNC PHP_FUNCTION(datefmt_format_object)
 					"not initialized properly", 0);
 			RETURN_FALSE;
 		}
-		timeZone = obj_cal->getTimeZone().clone();
+		timeZone = std::unique_ptr<TimeZone>(obj_cal->getTimeZone().clone());
 		date = obj_cal->getTime(status);
 		if (U_FAILURE(status)) {
 			intl_error_set(NULL, status,
 					"datefmt_format_object: error obtaining instant from "
 					"IntlCalendar", 0);
-			RETVAL_FALSE;
-			goto cleanup;
+			RETURN_FALSE;
 		}
-		cal = obj_cal->clone();
+		cal = std::unique_ptr<Calendar>(obj_cal->clone());
 	} else if (instanceof_function(instance_ce, php_date_get_interface_ce())) {
-		if (intl_datetime_decompose(object, &date, &timeZone, NULL,
+		TimeZone *tz;
+		if (intl_datetime_decompose(object, &date, &tz, NULL,
 				"datefmt_format_object") == FAILURE) {
 			RETURN_FALSE;
 		}
-		cal = new GregorianCalendar(Locale::createFromName(locale_str), status);
+		timeZone = std::unique_ptr<TimeZone>(tz);
+		cal = std::unique_ptr<Calendar>(new GregorianCalendar(Locale::createFromName(locale_str), status));
```

Comment:
Can't you use make_unique here instead of unique_ptr+new ?

Reply:
I wish isn't it C++14 minimum ?

Reply:
Ah right, and it's the ICU version that decides between C++11/C++17... Nvm then ;)